File: 14-08/multiMatriz.py

# Entrada do número de linhas e de colunas das matrizes
nLinhas = int(input("Digite o numero de linhas das matrizes: "))
nColunas = int(input("Digite o numero de colunas das matrizes: "))

# Declaração das matrizes
matriz1 = []
matriz2 = []

# Entrada de dados da matriz1
for i in range(nLinhas):
    linha = []
    for j in range(nColunas):
        valor = float(input(f"Digite o elemento [{i}][{j}] da matriz1: "))
        linha.append(valor)
    matriz1.append(linha)

# Entrada de dados da matriz2
for i in range(nLinhas):
    linha = []
    for j in range(nColunas):
        valor = float(input(f"Digite o elemento [{i}][{j}] da matriz2: "))
        linha.append(valor)
    matriz2.append(linha)

# Mostrando as matrizes
print("Matriz 1:")
for linha in matriz1:
    print(linha)

print("Matriz 2:")
for linha in matriz2:
    print(linha)
    
# Entrada de qual linha e coluna devem ser multiplicadas 
linhaMult = int(input('Digite a linha da matriz A que deseja multiplicar: '))
colunaMult = int(input('Digite a coluna da matriz B que deseja multiplicar: '))

# multiplicação da linha e da coluna, com validação se existe a linha e a coluna que vão ser multiplicadas
if linhaMult < 1 or linhaMult > nLinhas or colunaMult < 1 or colunaMult > nColunas:
    print("Linha ou coluna fora dos limites!")
else:
    res = 0
    for i in range(nLinhas):  
        res += matriz1[linhaMult-1][i] * matriz2[i][colunaMult-1]  

    print(f"O resultado é: {res}")

File: 14-08/multiMatriz.c

#include <stdio.h>

int main(){

	// Declaração de variaveis
	int nLinhas, nColunas, linhaMult, colunaMult;
	
	// Entrada do numero de linhas e de colunas das matrizes
	printf("Digite o numero de linhas das matrizes: ");
	scanf("%d", &nLinhas);
	printf("Digite o numero de colunas das matrizes: ");
	scanf("%d", &nColunas);
	
    // Declaração das matrizes
	double matriz1[nLinhas][nColunas];
	double matriz2[nLinhas][nColunas];
	double res = 0;

    // Entrada de dados das matrizes
	for(int i = 0; i<nLinhas; i++){
		for(int j = 0; j<nColunas;j++){
			printf("Digite o elemento [%d][%d] da matriz1: ", i, j);
			scanf("%lf", &matriz1[i][j]);
		}
	}

	for(int i = 0; i<nLinhas; i++){
		for(int j = 0; j<nColunas;j++){
			printf("Digite o elemento [%d][%d] da matriz2: ", i, j);
			scanf("%lf", &matriz2[i][j]);
		}
	}

    // Escolha de qual linha e qual coluna devem ser multiplicadas
	printf("Escolha uma das linhas da primeira matriz: ");
	scanf("%d", &linhaMult);

	printf("Escolha uma das colunas da segunda matriz: ");
	scanf("%d", &colunaMult);

    // Calculo da multiplicação dos elementos da linha da matriz A com uma coluna da Matriz B
	for(int i = 0; i < nLinhas; i++){
		res = res + matriz1[linhaMult-1][i] * matriz2[i][colunaMult-1]; 
	}
    
    // Exibição do resultado
	printf("O resultado é: %.2lf", res);

}

